include/gridtools/common/for_each.hpp

```diff
@@ -53,8 +54,11 @@ namespace gridtools {
 #endif
         } // namespace for_each_impl_
 
-        template <class L>
-        constexpr for_each_impl_::for_each_f<L> for_each = {};
+        template <typename L, typename Args>
```

Comment:
Please put the workaround into cuda `#ifdef`
- change `typename` to `class` for consistency
- change plural `Args` to the singular `Arg`